### PR TITLE
Support relative path in SYS_execve

### PR DESCRIPTION
**Pull request by asvrada**

### CODE REVIEW

**Comment by asvrada** [comment]
Passing cpython relative path tests

```
test_cwd_with_absolute_arg (Lib.test.test_subprocess.ProcessTestCase) ... ok
test_cwd_with_bytes (Lib.test.test_subprocess.ProcessTestCase) ... ok
test_cwd_with_pathlike (Lib.test.test_subprocess.ProcessTestCase) ... ok
test_cwd_with_relative_arg (Lib.test.test_subprocess.ProcessTestCase) ... ok
test_cwd_with_relative_executable (Lib.test.test_subprocess.ProcessTestCase) ... ok
test_double_close_on_error (Lib.test.test_subprocess.ProcessTestCase) ... ok
```

File: kernel/syscall.c [hunk 1]
```diff
@@ -2087,11 +2088,23 @@ long myst_syscall_execve(
...
+    // Resolve relative path
+    if (filename[0] != '/')
+    {
+        if (!(resolved_path = (myst_path_t*)malloc(sizeof(myst_path_t))))
+            ERAISE(-ENOMEM);
+
+        if (ret = myst_realpath(filename, resolved_path))
+            ERAISE(ret);
+
+        filename = resolved_path->buf;
```
Comment (mingweishih) [inline kernel/syscall.c:2105]: Suggest declare a new variable for this. Reusing a `const` parameter is not a good practice.

File: kernel/syscall.c [hunk 2]
```diff
@@ -2087,11 +2088,23 @@ long myst_syscall_execve(
...
+    if (!filename)
         ERAISE(-EINVAL);
 
+    // Resolve relative path
+    if (filename[0] != '/')
+    {
+        if (!(resolved_path = (myst_path_t*)malloc(sizeof(myst_path_t))))
+            ERAISE(-ENOMEM);
+
+        if (ret = myst_realpath(filename, resolved_path))
```
Comment (mingweishih) [inline kernel/syscall.c:2102]: Use `ECHECK(myst_realpath(filename, resolved_path))` instead

File: kernel/syscall.c [hunk 3]
```diff
@@ -2087,11 +2088,23 @@ long myst_syscall_execve(
     long ret = 0;
     const char** argv = NULL;
     myst_thread_t* current_thread = myst_thread_self();
+    myst_path_t* resolved_path = NULL;
```
Comment (mingweishih) [inline kernel/syscall.c:2091]: `myst_path_t` is a struct with a char array. You can just declare the variable instead of a pointer.

Comment (asvrada) [reply kernel/syscall.c:2091]: It would breach stack size limit if we do that

`error: stack usage might be 4272 bytes [-Werror=stack-usage=]`

Comment (mingweishih) [reply kernel/syscall.c:2091]: Not sure if it's configurable, can you check? 4272 bytes does not seem to be large to me though.

File: kernel/syscall.c (hunk 1 above)
Comment (asvrada) [reply kernel/syscall.c:2105]: Adding a new variable might be too confusing? How do you think about this way? No reusing `filename` and no new variable

```
argv[0] = resolved_path ? resolved_path->buf : filename;
```

Comment (mingweishih) [reply kernel/syscall.c:2105]: Would it be conflicted with the logic below? A simple way to just declare
`const char* path = filename;`
And use `path` throughout the function instead.

File: kernel/syscall.c (hunk 3 above)
Comment (asvrada) [reply kernel/syscall.c:2091]: The error could be ignored with `#pragma GCC diagnostic ignored "-Wstack-usage="`, but I am not sure if its a good idea, and couldn't find the place to configure the threshold... Do you have more info on this issue?

Comment (mingweishih) [reply kernel/syscall.c:2091]: ok, I see we set the limit to 512: https://github.com/deislabs/mystikos/blob/6cdeb80aef0faf7e807b953c44d241f3543a5676/kernel/Makefile#L42

Let's keep the current implementation.

**Review by mingweishih — APPROVED** [review]

**Pull request merged**